hostplumber/controllers/hostconfig_controller.go

```diff
@@ -0,0 +1,486 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	plumberv1 "hostplumber/api/v1"
+	hoststate "hostplumber/pkg/hoststate"
+)
+
+const (
+	sysClassNet   = "/host/sys/class/net/"
+	sysPciDrivers = "/host/sys/bus/pci/drivers/"
+	sysPciDevices = "/host/sys/bus/pci/devices/"
+)
+
+// HostConfigReconciler reconciles a HostConfig object
+type HostConfigReconciler struct {
+	client.Client
+	Log         logr.Logger
+	Scheme      *runtime.Scheme
+	NodeName    string
+	currentSpec *plumberv1.HostConfigSpec
```

Comment:
Same as in the other controller, try avoid storing object-specific state in the reconciler